cli/tests/functional/src/test/java/io/helidon/tests/functional/TestUtils.java

```diff
@@ -0,0 +1,164 @@
+                    .readTimeout(100*60*1000)
+                    .open();
+            ReadableByteChannel readableByteChannel = Channels.newChannel(is);
+            FileOutputStream fileOutputStream = new FileOutputStream(Files.createFile(destination).toString());
+            fileOutputStream.getChannel().transferFrom(readableByteChannel, 0, Long.MAX_VALUE);
+            fileOutputStream.close();
+        } catch (IOException e) {
+            throw new UncheckedIOException("Download failed at URL : " + url, e);
+        }
+    }
+
+    static void waitForApplication(int port) throws Exception {
+        long timeout = 60 * 1000;
+        long now = System.currentTimeMillis();
+        URL url = new URL("http://localhost:" + port + "/greet");
+
+        HttpURLConnection conn = null;
+        int responseCode;
+        do {
+            Thread.sleep(1000);
+            if ((System.currentTimeMillis() - now) > timeout) {
+                throw new Exception("Application failed to start on port :" + port);
+            }
+            try {
+                conn = (HttpURLConnection) url.openConnection();
+                conn.setConnectTimeout(500);
+                responseCode = conn.getResponseCode();
+            } catch (Exception ex) {
+                responseCode = -1;
+            }
+            if (conn != null) {
+                conn.disconnect();
+            }
+        } while (responseCode != 200);
+    }
+
+    static int getAvailablePort() {
+        try {
+            ServerSocket s = new ServerSocket(0);
+            s.close();
+            return s.getLocalPort();
+        } catch (IOException ioe) {
+            throw new UncheckedIOException(ioe);
+        }
+    }
+
+    static void generateBareSe(Path wd, String mavenHome) {
+        List<String> mavenArgs = List.of(
+                "archetype:generate",
+                "-DinteractiveMode=false",
+                "-DarchetypeGroupId=io.helidon.archetypes",
+                "-DarchetypeArtifactId=helidon",
+                "-DarchetypeVersion=3.0.0-M1",
+                "-DgroupId=groupid",
+                "-DartifactId=artifactid",
+                "-Dpackage=custom.pack.name",
+                "-Dflavor=se",
+                "-Dbase=bare");
+
+        try {
+            MavenCommand.builder()
+                    .executable(Path.of(mavenHome, "apache-maven-3.8.4", "bin", TestUtils.getMvnExecutable("3.8.4")))
+                    .directory(wd)
+                    .addArguments(mavenArgs)
+                    .build()
+                    .execute();
+        } catch (Exception e) {
+            throw new IllegalStateException("Cannot generate bare-se project", e);
+        }
+    }
+
+    static String getMvnExecutable(String mavenVersion) {
+        if (OSType.currentOS().equals(OSType.Windows)) {
+            if (MavenVersion.toMavenVersion(mavenVersion).isLessThanOrEqualTo(MavenVersion.toMavenVersion("3.2.5"))) {
+                return "mvn.bat";
```

Comment:
Wouldn't it be more robust to test the existence if `.cmd` or `.bat` if on windows, regardless of the Maven version ?